services/web/server/src/simcore_service_webserver/users/_notifications.py

```diff
@@ -30,6 +31,7 @@ class BaseUserNotification(BaseModel):
     title: str
     text: str
     date: datetime
+    product: Literal["UNDEFINED"] | ProductName
```

Comment:
I would go for this here. It's also set by default to None, which is what you had for older notifications. This allows you to properly be compatible with older data
```suggestion
    product: ProductName | None = None
```

Reply:
I was recommended to go for Literal["UNDEFINED"] instead of None